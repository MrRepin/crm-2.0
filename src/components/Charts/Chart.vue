<template>
    <v-container fluid class="chart">
        <v-row>
            <v-col class="md-12">
                <v-card class="chart-card">
                    <v-row>
                        <v-col class="md-6">
                            <ColorIcon icon="mdi-earth" />
                            <p class="text-left title font-weight-light">Total sales by country</p>
                            <DefaultTable :tableData="countries" />
                        </v-col>
                        <v-col class="md-6">
                            <canvas id="chart" ref="canvas"></canvas>
                        </v-col>
                    </v-row>
                </v-card>
            </v-col>
        </v-row>
    </v-container>
</template>

<script>
import {Doughnut} from 'vue-chartjs'
import DefaultTable from '@/components/Tables/DefaultTable'
import ColorIcon from '@/components/Icons/ColorIcon'
export default {
    name: 'GlobalMap',
    extends: Doughnut,
    props: {
        chart: Object
    },
    mounted () {
        this.renderChart(this.chart)
    },
    data: () => ({
        countries: [
            {
                name: 'Brasil',
                quant: 3890,
                flag: require('@/assets/img/brasil.png'),
                color: 'red'
            },
            {
                name: 'USA',
                quant: 2540,
                flag: require('@/assets/img/usa.png'),
                color: 'blue'
            },
            {
                name: 'Australia',
                quant: 1820,
                flag: require('@/assets/img/australia.png'),
                color: 'green'
            },
            {
                name: 'Germany',
                quant: 1220,
                flag: require('@/assets/img/germany.png'),
                color: 'black'
            },
            {
                name: 'UK',
                quant: 834,
                flag: require('@/assets/img/uk.png'),
                color: 'green'
            },
            {
                name: 'Romania',
                quant: 356,
                flag: require('@/assets/img/romania.png'),
                color: 'yellow'
            }
        ],
    }),
    components: {
        DefaultTable,
        ColorIcon
    }
}
</script>
<style lang="sass" scoped>
    .title
        font-size: 20px
        margin-left: 84px

    .chart
        padding: 0
        margin-top: 30px

    .chart-card
        padding: 15px
        position: relative
</style>