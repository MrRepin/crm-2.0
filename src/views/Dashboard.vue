<template>
    <div id="dashboard">
        <InfoCards :cards="infoCards" />
        <Chart :chart="chart" />
        <StatusCards :cards="statusCards" />
    </div>
</template>

<script>
import InfoCards from '@/components/Cards/InfoCards'
import Chart from '@/components/Charts/Chart'
import StatusCards from '@/components/Cards/StatusCards'
export default {
    name: "Dashboard",
    data: () => ({
        chart: {
            labels: ['Brasil', 'USA', 'Australia', 'Germany', 'UK'],
            datasets: [{
                responsive: true,
                label: '# of Votes',
                data: [2540, 1820, 1220, 834, 356],
                backgroundColor: [
                    'green',
                    'red',
                    'blue',
                    'purple',
                    'orange',
                    'yellow',
                ],
            }]
        },
        infoCards: [
            {
                name: 'Orders',
                icon: 'mdi-shopping-outline',
                info: '2134',
                color: 'blue'
            },
            {
                name: 'Sales',
                icon: 'mdi-currency-usd',
                info: '657',
                color: 'green'
            },
            {
                name: 'Returns',
                icon: 'mdi-repeat',
                info: '21',
                color: 'red'
            },
            {
                name: 'Messages',
                icon: 'mdi-forum',
                info: '329',
                color: 'orange'
            }
        ],
        statusCards: [
            {
                background: '#2192ed',
                data: {
                    name: 'Daily sales',
                    chart: {
                        labels: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
                        datasets: [{
                            responsive: true,
                            label: 'Daily sales',
                            data: [250, 341, 213, 240, 250, 300, 1200],
                            borderColor: '#fff',
                        }]
                    }
                },
            },
            {
                background: '#4caf50',
                data: {
                    name: 'Daily sales',
                    chart: {
                        labels: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
                        datasets: [{
                            responsive: true,
                            label: 'Daily sales',
                            data: [250, 341, 213, 240, 250, 300, 1200],
                            borderColor: '#fff'
                        }]
                    },
                },
            },
        ]
    }),
    components: {
        InfoCards,
        Chart,
        StatusCards
    }
}
</script>