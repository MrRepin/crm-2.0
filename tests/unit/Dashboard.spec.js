import Vue from 'vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import * as VueChart from 'vue-chartjs'
import Chart from '@/components/Charts/Chart'

import Dashboard from '@/views/Dashboard'

Vue.use(Vuetify)

const localVue = createLocalVue()
localVue.use(VueChart)
localVue.use(Chart)

describe('Dashboard', () => {
  const wrapper = shallowMount(Dashboard, {
    localVue
  })
  it('is a Vue instance', () => {
    expect(wrapper.isEmpty()).toBeFalsy()
  })
  it('is render correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})