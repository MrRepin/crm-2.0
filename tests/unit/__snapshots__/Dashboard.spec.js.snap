// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard is render correctly 1`] = `
<div
  id="dashboard"
>
  <infocards-stub
    cards="[object Object],[object Object],[object Object],[object Object]"
  />
   
  <chart-stub
    chart="[object Object]"
    chartid="doughnut-chart"
    cssclasses=""
    height="400"
    plugins=""
    width="400"
  />
   
  <statuscards-stub
    cards="[object Object],[object Object]"
  />
</div>
`;
